src/main/java/com/iridium/iridiumskyblock/managers/tablemanagers/ForeignIslandTableManager.java

```diff
@@ -26,18 +26,19 @@ public ForeignIslandTableManager(ConnectionSource connectionSource, Class<T> cla
 
     @Override
     public void addEntry(T t) {
-        getEntries().add(t);
+        int index = Collections.binarySearch(getEntries(), t, Comparator.comparing(t1 -> t1.getIsland().map(Island::getId).orElse(0)));
+        getEntries().add(index < 0 ? -(index + 1) : index, t);
     }
 
     /**
      * Sort the list of entries by island id
      */
     private void sort() {
-        getEntries().sort(comparator);
+        getEntries().sort(Comparator.comparing(t1 -> t1.getIsland().map(Island::getId).orElse(0)));
```

Comment:
This should be sorted using comparator? In Rewards case comparator is sorting by IslandID (which you can see in DatabaseManager)